apps/studio/prisma/scripts/migrations/addDefaultIndexPage/__tests__/addDefaultIndexPage.test.ts

```diff
@@ -0,0 +1,185 @@
+// This migration is to update the Folders that currently
+// do not have an `IndexPage` to have a default one.
+// It is a one-time migration to update the existing Folder records
+// It is not reversible
+
+import { ResourceType } from "~prisma/generated/generatedEnums"
+import { resetTables } from "tests/integration/helpers/db"
+import {
+  setupCollection,
+  setupFolder,
+  setupPageResource,
+  setupSite,
+} from "tests/integration/helpers/seed"
+
+import { db, jsonb } from "~/server/modules/database"
+import { createFolderIndexPage } from "~/server/modules/page/page.service"
+import { createDefaultFolderIndexPage } from "../addDefaultIndexPage"
+
+describe("createDefaultFolderIndexPage", () => {
+  let site: Awaited<ReturnType<typeof setupSite>>["site"]
+
+  beforeEach(async () => {
+    await resetTables("Blob", "Resource")
+  })
+
+  beforeAll(async () => {
+    const { site: _site } = await setupSite()
+    site = _site
+  })
+
+  it("should not impact folders with existing index pages", async () => {
+    // Arrange
+    const { folder } = await setupFolder({ siteId: site.id })
+    const { page } = await setupPageResource({
+      parentId: folder.id,
+      siteId: site.id,
+      resourceType: ResourceType.IndexPage,
+    })
+
+    // Act
+    await createDefaultFolderIndexPage()
+
+    // Assert
+    const indexPage = await getIndexPageOf(folder.id)
+    expect(indexPage).toEqual(page)
```

Comment:
this should not matter as much since this is for test purposes, but IMO since `getIndexPageOf` is using  `executeTakeFirst()`, which means even if an index page is indeed wrongfully created, this test might still pass if the first result returned is the same.

Since we are checking that folders with existing index pages should not be impacted, perhaps its better to use `execute()` and also assert that the total number of indexpage for this folder is the same before-after (which is 1)